src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonersResourceIntTest.kt

```diff
@@ -22,6 +24,173 @@ class PrisonersResourceIntTest : IntegrationTestBase() {
+          Tuple(oldBooking.bookingId, rootOffender.id, rootOffender.id, 2, false),
+          Tuple(aliasBooking.bookingId, aliasOffender1.id, rootOffender.id, 3, false),
+          Tuple(oldAliasBooking.bookingId, aliasOffender1.id, rootOffender.id, 4, false),
+        )
+    }
+    with(aliasOffender1) {
+      assertThat(rootOffenderId).isEqualTo(rootOffender.id)
+      assertThat(getAllBookings()).extracting("bookingId")
+        .containsExactlyInAnyOrder(
+          activeBooking.bookingId,
+          oldBooking.bookingId,
+          aliasBooking.bookingId,
+          oldAliasBooking.bookingId,
+        )
+    }
+    with(aliasOffender2) {
+      assertThat(rootOffenderId).isEqualTo(rootOffender.id)
+      assertThat(getAllBookings()).extracting("bookingId")
+        .containsExactlyInAnyOrder(
+          activeBooking.bookingId,
+          oldBooking.bookingId,
+          aliasBooking.bookingId,
+          oldAliasBooking.bookingId,
+        )
+    }
+
+    // assert that entities loaded from database are correct
+
+    val root = repository.getOffender(nomsId = "A1234TT")!!
+    val alias1 = repository.getOffender(aliasOffender1.id)!!
+    val alias2 = repository.getOffender(aliasOffender2.id)!!
+    with(root) {
+      assertThat(id).isEqualTo(rootOffender.id)
+      assertThat(getAllBookings()).extracting("bookingId", "offender.id", "rootOffender.id", "bookingSequence", "active")
+        .containsExactlyInAnyOrder(
+          Tuple(activeBooking.bookingId, rootOffender.id, rootOffender.id, 1, true),
+          Tuple(oldBooking.bookingId, rootOffender.id, rootOffender.id, 2, false),
+          Tuple(aliasBooking.bookingId, aliasOffender1.id, rootOffender.id, 3, false),
+          Tuple(oldAliasBooking.bookingId, aliasOffender1.id, rootOffender.id, 4, false),
+        )
+      assertThat(getAllBookings()).extracting("bookingId")
+        .containsExactlyInAnyOrder(
+          activeBooking.bookingId,
+          oldBooking.bookingId,
+          aliasBooking.bookingId,
+          oldAliasBooking.bookingId,
+        )
+    }
+    with(alias1) {
+      assertThat(id).isEqualTo(aliasOffender1.id)
+      assertThat(rootOffenderId).isEqualTo(rootOffender.id)
+      assertThat(getAllBookings()).extracting("bookingId")
+        .containsExactlyInAnyOrder(
+          activeBooking.bookingId,
+          oldBooking.bookingId,
+          aliasBooking.bookingId,
+          oldAliasBooking.bookingId,
+        )
+    }
+    with(alias2) {
+      assertThat(id).isEqualTo(aliasOffender2.id)
+      assertThat(rootOffenderId).isEqualTo(rootOffender.id)
+      assertThat(getAllBookings()).extracting("bookingId")
+        .containsExactlyInAnyOrder(
+          activeBooking.bookingId,
+          oldBooking.bookingId,
+          aliasBooking.bookingId,
+          oldAliasBooking.bookingId,
+        )
+    }
+  }
+
+  @Test
+  fun `DSL setup check - latest booking on alias`() {
```

Comment:
Can we have another test - where there is more than 1 root offender (with aliases and bookings on each)? I'm 99% sure it's fine, but that neat trick of passing the OffenderBuilder into the AliasBuilder means there's some state shared so would be good to make sure it scales up.